src/main/java/com/sapozhnykov/dao/cart/impl/CartDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.sapozhnykov.dao.cart.impl;
+
+import com.sapozhnykov.dao.cart.CartDao;
+import com.sapozhnykov.domain.Product;
+
+import java.util.ArrayList;
+
+public class CartDaoImpl implements CartDao {
+    private ArrayList<Product> cartProducts = new ArrayList<>();
+
+    @Override
+    public boolean add(Product product) {
+        return cartProducts.add(product);
+    }
+
+    @Override
+    public boolean deleteById(String id) {
+        Product tempProduct;
+
+        tempProduct = getById(id);
+        return cartProducts.remove(tempProduct);
+    }
+
+    @Override
+    public ArrayList<Product> getAll() {
+        return this.cartProducts;
+    }
+
+    @Override
+    public Product getById(String id) {
+        long tempId;
+
+        try {
+            tempId = Long.parseLong(id);
```

Comment:
DAO should only work with storage and not processed data

Reply:
At the last lesson, Artem said that we would not have a Cart. I removed it from the project.